src/Evolution/DgSubcell/GhostZoneLogicalCoordinates.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Evolution/DgSubcell/SliceTensor.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace evolution::dg::subcell::fd {
+
+template <size_t Dim>
+void ghost_zone_logical_coordinates(
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::ElementLogical>*>
+        ghost_logical_coords,
+    const Mesh<Dim>& subcell_mesh, const size_t ghost_zone_size,
+    const Direction<Dim>& direction) {
+  const size_t dim_direction = direction.dimension();
+  const auto subcell_extents = subcell_mesh.extents();
+
+  // Check if the `ghost_zone_size` has a valid value.
+  ASSERT(ghost_zone_size <= subcell_extents[dim_direction],
+         " Ghost zone size (" << ghost_zone_size
+                              << ") is larger than the volume extent ("
+                              << subcell_extents[dim_direction] << ")");
+
+  const auto subcell_logical_coords = logical_coordinates(subcell_mesh);
```

Comment:
Sorry, I thought the implementation we discussed doesn't do this slicing. Slicing is surprisingly expensive, I think we should just calculate the points explicitly